src/main/java/ch/njol/skript/expressions/arithmetic/ExprArithmetic.java

```diff
@@ -117,43 +118,96 @@ public PatternInfo(Operator operator, boolean leftGrouped, boolean rightGrouped)
 	private boolean leftGrouped, rightGrouped;
 
 	@Override
-	@SuppressWarnings({"ConstantConditions", "unchecked"})
+	@SuppressWarnings({"ConstantConditions", "rawtypes", "unchecked"})
 	public boolean init(Expression<?>[] exprs, int matchedPattern, Kleenean isDelayed, ParseResult parseResult) {
-		first = LiteralUtils.defendExpression(exprs[0]);
-		second = LiteralUtils.defendExpression(exprs[1]);
-
-		if (!LiteralUtils.canInitSafely(first, second))
-			return false;
-
-		Class<? extends L> firstClass = first.getReturnType();
-		Class<? extends R> secondClass = second.getReturnType();
+		first = (Expression<L>) exprs[0];
+		second = (Expression<R>) exprs[1];
 
 		PatternInfo patternInfo = patterns.getInfo(matchedPattern);
 		leftGrouped = patternInfo.leftGrouped;
 		rightGrouped = patternInfo.rightGrouped;
 		operator = patternInfo.operator;
 
-		if (firstClass != Object.class && secondClass == Object.class && Arithmetics.getOperations(operator, firstClass).isEmpty()) {
-			// If the first class is known but doesn't have any operations, then we fail
-			return error(firstClass, secondClass);
-		} else if (firstClass == Object.class && secondClass != Object.class && Arithmetics.getOperations(operator).stream()
-				.noneMatch(info -> info.getRight().isAssignableFrom(secondClass))) {
-			// If the second class is known but doesn't have any operations, then we fail
-			return error(firstClass, secondClass);
+		// try to convert unparsed literals with known info
```

Comment:
what are the chances we can move this to a private method

Reply:
any specific reason? it has to update the fields so i think keeping it here improves the readability

Reply:
i'm just worried we are attracting bugs by having a 100+ line init method

Reply:
after a lot of thinking, i have elected to keep it as one method, but i have greatly expanded the documentation to ensure the process is clearly outlined.